tests/common.go

```diff
@@ -1904,6 +1912,61 @@ func ValidateApplications(contexts []*scheduler.Context) {
 	})
 }
 
+// ValidateApplications validates applications
+func ValidateApplicationsStartData(contexts []*scheduler.Context, appContext context1.Context) (chan string, *errgroup.Group, map[string][]appDriver.ApplicationDriver) {
+
+	// Resetting the global map before starting the new App Validations
+	NamespaceAppWithDataMap = make(map[string][]appDriver.ApplicationDriver)
+
+	var allHandlers []appDriver.ApplicationDriver
+
+	Step("validate applications", func() {
+		log.InfoD("Validate applications")
+		for _, ctx := range contexts {
+			ValidateContext(ctx)
+			appInfo, err := appUtils.ExtractConnectionInfo(ctx)
+			if err != nil {
+				log.InfoD("Some error occurred - [%s]", err)
+			}
+			log.InfoD("App Info - [%+v]", appInfo)
+			if appContext == nil {
+				log.Warnf("App Context is not proper - [%v]", appContext)
+				continue
+			}
+			if appInfo.StartDataSupport {
+				appHandler, _ := appDriver.GetApplicationDriver(
+					appInfo.AppType,
+					appInfo.Hostname,
+					appInfo.User,
+					appInfo.Password,
+					appInfo.Port,
+					appInfo.DBName,
+					appContext,
+					appInfo.NodePort,
+					appInfo.Namespace)
+				log.InfoD("App handler created for [%s]", appInfo.Hostname)
+				NamespaceAppWithDataMap[appInfo.Namespace] = append(NamespaceAppWithDataMap[appInfo.Namespace], appHandler)
+				allHandlers = append(allHandlers, appHandler)
```

Comment:
Why are adding appHandlers to `NamespaceAppWithDataMap` and also `allHandlers` ?

Reply:
It was added earlier when NamespaceAppWithDataMap was not present, will modify the same and only use NamespaceAppWithDataMap to start the routines.

Reply:
Done